functions/auth.js

```diff
@@ -225,3 +226,25 @@ exports.connectI4AOAuthHandler = functions.https.onRequest(async (req, res) => {
 
   res.redirect(customTokenReturnUrl.toString());
 });
+
+/** Automatically make user admin upon register.
+ *
+ *  A function that triggers when a Firebase user is created, not on https request
+ *
+ *  Firebase accounts will trigger user creation events for Cloud Functions when:
+ *    - A user creates an email account and password.
+ *    - A user signs in for the first time using a federated identity provider.
+ *    - The developer creates an account using the Firebase Admin SDK.
+ *    - A user signs in to a new anonymous auth session for the first time.
+ *
+ *  NOTE: A Cloud Functions event is not triggered when a user signs in for the first time using a custom token.
+ *
+ *  @see https://firebase.google.com/docs/functions/auth-events
+ */
+exports.autoAdminOnRegister = functions.auth.user().onCreate(async (user) => {
+  const flag = functions.config().flag || {};
```

Comment:
[NB] using guard ` || {} ` as to not force older environments retrofit to have a flag object until necessary in their respective cases